apps/prairielearn/assets/scripts/workspaceClient.ts

```diff
@@ -89,7 +92,16 @@ $(function () {
       workspaceFrame.src = 'about:blank';
       if (previousState === 'running') {
         showStoppedFrame();
-      } else {
+      } else if (previousState !== 'uninitialized') {
+        // When the workspace is first created, it will be in the `uninitialized`
+        // state. It then transitions to the `stopped` state, and then immediately
+        // to `launching`.
+        //
+        // We don't want to consider the initial transition to `stopped` as a
+        // failure, so we specifically ignore transitions from `uninitialized`.
+        //
+        // Put differently: we'll really only show the failure message if we
+        // transition directly from `launching` to `stopped`.
```

Comment:
We could also write the above conditional as `previousState === 'launching'`. I don't have a strong preference either way.

Reply:
I'm fine either way as well. Ironically my initial implementation also had `previousState === 'launching'`, but I changed it to make it more "generic"...